packages/camera/camera_windows/windows/capture_controller.cpp

```diff
@@ -237,14 +237,16 @@ HRESULT CaptureControllerImpl::CreateCaptureEngine() {
     return hr;
   }
 
+  // Check MF_CAPTURE_ENGINE_INITIALIZED event handling
+  // for response process.
   hr = capture_engine_->Initialize(capture_engine_callback_handler_.Get(),
                                    attributes.Get(), audio_source_.Get(),
                                    video_source_.Get());
   return hr;
 }
 
 void CaptureControllerImpl::ResetCaptureController() {
-  if (record_handler_) {
+  if (record_handler_ && record_handler_->CanStop()) {
```

Comment:
Previously `IsContinuousRecording` and `IsTimedRecording` could return `true` even if the recording has stopped. This caused the destructor to stop recording even though recording was already stopped.